code/drasil-gool/GOOL/Drasil/LanguageRenderer/CLike.hs

```diff
@@ -120,26 +121,30 @@ listSize v = v $. S.listSizeFunc
 -- Statements --
 
 increment1 :: (RenderSym r) => SVariable r -> MSStatement r
-increment1 vr = zoom lensMStoVS $ onStateValue (mkStmt . R.increment) vr
+increment1 vr' = on1StateWrapped (mkStmt . R.increment) $ 
+  zoom lensMStoVS vr'
 
 varDec :: (RenderSym r) => r (Permanence r) -> r (Permanence r) -> Doc -> 
   SVariable r -> MSStatement r
-varDec s d pdoc v' = onStateValue (\v -> mkStmt (RC.perm (bind $ variableBind v)
-  <+> RC.type' (variableType v) <+> (ptrdoc (getType (variableType v)) <> 
-  RC.variable v))) (zoom lensMStoVS v')
+varDec s d pdoc v' = do 
+  v <- zoom lensMStoVS v' 
+  mkStmt (RC.perm (bind $ variableBind v)
+    <+> RC.type' (variableType v) <+> (ptrdoc (getType (variableType v)) <> 
+    RC.variable v))
   where bind Static = s
         bind Dynamic = d
         ptrdoc (List _) = pdoc
         ptrdoc _ = empty
 
 varDecDef :: (RenderSym r) => SVariable r -> SValue r -> MSStatement r
-varDecDef vr vl' = on2StateValues (\vd vl -> mkStmt (RC.statement vd <+> equals 
-  <+> RC.value vl)) (S.varDec vr) (zoom lensMStoVS vl')
+varDecDef vr vl' = on2StateWrapped (\vd vl -> mkStmt 
+  (RC.statement vd <+> equals <+> RC.value vl)) (S.varDec vr) $ 
+  zoom lensMStoVS vl'
 
 listDec :: (RenderSym r) => (r (Value r) -> Doc) -> SValue r -> SVariable r -> 
   MSStatement r
-listDec f vl v = on2StateValues (\sz vd -> mkStmt (RC.statement vd <> f 
-  sz)) (zoom lensMStoVS vl) (S.varDec v)
+listDec f vl v = on2StateWrapped (\sz vd -> mkStmt (RC.statement vd <> f sz))
```

Comment:
I think a lot of `on2StateWrapped` would also be better using `do` notation. This one for sure.